pool/app/logger/logger.ml

```diff
@@ -118,9 +118,99 @@ let format_reporter
   { Logs.report }
 ;;
 
+let get_log_level () =
+  match Sys.getenv_opt "LOG_LEVEL" with
+  | Some "debug" -> Some Logs.Debug
+  | Some "error" -> Some Logs.Error
+  | Some "warning" -> Some Logs.Warning
+  | _ -> Some Logs.Info
+;;
+
+let logs_dir () =
+  match
+    Sihl.Configuration.root_path (), Sihl.Configuration.read_string "LOGS_DIR"
+  with
+  | _, Some logs_dir -> logs_dir
+  | Some root, None -> root ^ "/logs"
+  | None, None -> "logs"
+;;
+
+let lwt_file_reporter ?(pp_header = pp_exec_header) () =
+  let logs_dir = logs_dir () in
+  let buf () =
+    let b = Buffer.create 512 in
+    ( b
+    , fun () ->
+        let m = Buffer.contents b in
+        Buffer.reset b;
+        m )
+  in
+  let app, app_flush = buf () in
+  let err, err_flush = buf () in
+  let report src level ~over k msgf =
+    let k _ = k () in
+    let write () =
+      let name =
+        match[@warning "-4"] level with
+        | Logs.Error -> logs_dir ^ "/error.log"
+        | _ -> logs_dir ^ "/app.log"
+      in
+      let%lwt log =
+        Lwt_io.open_file
+          ~flags:[ Unix.O_WRONLY; Unix.O_CREAT; Unix.O_APPEND ]
+          ~perm:0o777
+          ~mode:Lwt_io.Output
+          name
+      in
+      let%lwt () =
+        match[@warning "-4"] level with
+        | Logs.Error -> Lwt_io.write log (err_flush ())
+        | _ -> Lwt_io.write log (app_flush ())
+      in
+      Lwt_io.close log
+    in
+    let unblock () =
+      over ();
+      Lwt.return_unit
+    in
+    Lwt.finalize write unblock |> Lwt.ignore_result;
```

Comment:
Use the `lwt.finally` ppx here please to ensure stack traces (https://github.com/uzh/pool/commit/7867e2c68906ced3199a66d388caa2bd48f85e0d#diff-1f214c71db6e40598f3df14bec1437e3e41f5359bcd83bfdcab6cfc527ebde48L114-R124 see here for example)

Reply:
Done